googletest/include/gtest/internal/gtest-internal.h

```diff
@@ -1337,17 +1346,24 @@ constexpr bool InstantiateTypedTestCase_P_IsDeprecated() { return true; }
 
 #define GTEST_TEST_NO_THROW_(statement, fail) \
   GTEST_AMBIGUOUS_ELSE_BLOCKER_ \
-  if (::testing::internal::AlwaysTrue()) { \
+  if (::testing::internal::TrueWithString gtest_msg = std::string()) { \
```

Comment:
Why can't this be declared on the next line?

Reply:
Same reason as in `GTEST_TEST_THROW_` - `gtest_msg` is used in both `if` and `else`

Reply:
Can this be written with `::testing::internal::ConstCharPtr` instead of introducing a new type?

Reply:
It cannot: we need to own a copy of a string. ConstCharPtr is non-owning a wrapper over `const char*`